examples/layers/raw/shader_warm_up.dart

```diff
@@ -9,14 +9,14 @@ import 'dart:ui' as ui;
 import 'package:flutter/material.dart';
 import 'package:flutter/painting.dart' show DefaultShaderWarmUp;
 
-void beginFrame(Duration timeStamp) {
+Future<void> beginFrame(Duration timeStamp) async {
```

Comment:
beginFrame should be void and sync. Making it async means this will not work correctly (it'll return before `render` is called). I'm surprised this example still works (if it does?).

Reply:
It still works with `flutter run raw/shader_warm_up.dart` inside `examples/layers`.

Sure we call `render` after `Future<void>` is returned. But I thought that's needed because `render` has to wait for `warmUpOnCanvas` to prepare the picture. Why do we have to ensure that `beginFrame` return after `render` is called?

If we need to do that, how are we going to wait for the `warmUpOnCanvas`? Someone told me that Dart is single-threaded and there's no way to wait for a future inside a sync function?